src/Microsoft.Data/DataFrameColumnCollection.cs

```diff
@@ -88,21 +88,22 @@ public void InsertColumn(int columnIndex, DataFrameColumn column)
             RowCount = column.Length;
             _columnNames.Insert(columnIndex, column.Name);
             _columnNameToIndexDictionary[column.Name] = columnIndex;
-            for (int i = columnIndex + 1; i < ColumnCount; i++)
+            for (int i = columnIndex + 1; i < Count; i++)
             {
                 _columnNameToIndexDictionary[_columnNames[i]]++;
             }
-            _columns.Insert(columnIndex, column);
-            ColumnCount++;
+            base.InsertItem(columnIndex, column);
+            ColumnsChanged?.Invoke();
         }
 
-        public void SetColumn(int columnIndex, DataFrameColumn column)
+        public void Set(int columnIndex, DataFrameColumn column)
```

Comment:
Why have this public API when a caller can just use the indexer?

Reply:
Made it internal